### PR TITLE
Add CONTRIBUTING.md and DEVELOPMENT.md and update README.md

### PR DESCRIPTION
**Pull request by natalieparellano**
Fixes https://github.com/buildpacks/lifecycle/issues/302 (mostly modeled after `pack`)

- Adds CI best practices badge
- Documents insecure registries workaround
- Adds basic information for understanding and contributing to lifecycle

Signed-off-by: Natalie Arellano <narellano@vmware.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+### Background
+
+Here are some topics that might be helpful in further understanding the lifecycle:
+
+* Cloud Native Buildpacks platform api spec
+  * Example platforms: [pack CLI](https://github.com/buildpack/pack), [Tekton](https://github.com/tektoncd/catalog/blob/master/task/buildpacks/0.1/README.md)
+* Cloud Native Buildpacks buildpack api spec
+  * Example buildpack providers: [Google](https://github.com/GoogleCloudPlatform/buildpacks), [Heroku](https://www.heroku.com/), [Paketo](https://paketo.io/)
+* The Open Container Initiative (OCI) and (OCI image spec](https://github.com/opencontainers/image-spec)
```
Comment (natalieparellano) [inline CONTRIBUTING.md:28]:
```suggestion
* The Open Container Initiative (OCI) and [OCI image spec](https://github.com/opencontainers/image-spec)
```

File: RELEASE.md [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
 Inside the `pack` repo:
-```
-LIFECYCLE_PATH="/path/to/created/lifecycle-v0.99.0+linux.x86-64.tgz" make acceptance
-
-$env:LIFECYCLE_PATH="C:\path\to\created\lifecycle-v0.99.0+windows.x86-64.tgz" 
-make acceptance
-```
+    On Linux:
+    ```
+    LIFECYCLE_PATH="/path/to/created/lifecycle-v0.99.0+linux.x86-64.tgz" make acceptance
```
Comment (jabrown85) [inline RELEASE.md:14]: Are these relevant now? Can we pass a specific lifecycle via a `pack` arg now?

**Review by jabrown85 — COMMENTED** [review]
I had one question, nothing blocking. This looks great!

**Review by jabrown85 — APPROVED** [review]

File: RELEASE.md (hunk 2 above)
Comment (natalieparellano) [reply RELEASE.md:14]: `--lifecycle-image` is a thing now, but not `--lifecycle`. But come to think of it, I should update the description to include the use of `LIFECYCLE_IMAGE="buildpacksio/lifecycle:<random-tag>"`.

Comment (natalieparellano) [inline RELEASE.md:14]:
```suggestion
    LIFECYCLE_PATH="/path/to/created/lifecycle-v0.99.0+linux.x86-64.tgz" LIFECYCLE_IMAGe="buildpacksio/lifecycle:<commit sha>" make acceptance
```

File: RELEASE.md [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
+    ```
+    On Windows:
+    ```
+    $env:LIFECYCLE_PATH="C:\path\to\created\lifecycle-v0.99.0+windows.x86-64.tgz"
```
Comment (natalieparellano) [inline RELEASE.md:18]:
```suggestion
    $env:LIFECYCLE_PATH="C:\path\to\created\lifecycle-v0.99.0+windows.x86-64.tgz" 
    $env:LIFECYCLE_IMAGE="buildpacksio/lifecycle:<commit sha>"
```

File: RELEASE.md [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
 Inside the `pack` repo:
-```
-LIFECYCLE_PATH="/path/to/created/lifecycle-v0.99.0+linux.x86-64.tgz" make acceptance
-
-$env:LIFECYCLE_PATH="C:\path\to\created\lifecycle-v0.99.0+windows.x86-64.tgz" 
-make acceptance
-```
+    On Linux:
+    ```
+    LIFECYCLE_PATH="/path/to/created/lifecycle-v0.99.0+linux.x86-64.tgz" LIFECYCLE_IMAGe="buildpacksio/lifecycle:<commit sha>" make acceptance
```
Comment (natalieparellano) [inline RELEASE.md:14]:
```suggestion
    LIFECYCLE_PATH="/path/to/created/lifecycle-v0.99.0+linux.x86-64.tgz" LIFECYCLE_IMAGE="buildpacksio/lifecycle:<commit sha>" make acceptance
```

File: RELEASE.md (hunk 2 above)
Comment (natalieparellano) [reply RELEASE.md]: Done!

**Pull request merged**